code/game/centcomm_orders/orders_medical.dm

```diff
@@ -0,0 +1,321 @@
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                                                                                          //
+//                                            MEDICAL ORDERS                                                //
+//                                                                                                          //
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//9 orders
+
+//----------------------------------------------Surgery----------------------------------------------------
+
+/datum/centcomm_order/department/medical
+	request_consoles_to_notify = list(
+		"Chief Medical Officer's Desk",
+		"Medbay",
+		)
+	acct_by_string = "Medical"
+
+/datum/centcomm_order/department/medical/kidneys/New()
+	..()
+	requested = list(
+		/obj/item/organ/internal/kidneys = rand(1,2)
+	)
+	name_override = list(
+		/obj/item/organ/internal/kidneys = "human kidneys"
+	)
+	extra_requirements = "The organs needs to be fresh, use a medical crate or a freezer."
+	worth = 200*requested[requested[1]]
+
+/datum/centcomm_order/department/medical/kidneys/ExtraChecks(var/obj/item/organ/internal/kidneys/I)
+	if (!istype(I))
+		return 0
+	if (I.health > 0)
+		return 1
+	return 0
+
+/datum/centcomm_order/department/medical/heart/New()
+	..()
+	requested = list(
+		/obj/item/organ/internal/heart = 1
+	)
+	name_override = list(
+		/obj/item/organ/internal/heart = "human heart"
+	)
+	extra_requirements = "The organ needs to be fresh, use a medical crate or a freezer."
+	worth = 400
+
+/datum/centcomm_order/department/medical/heart/ExtraChecks(var/obj/item/organ/internal/heart/I)
+	if (!istype(I))
+		return 0
+	if (I.health > 0)
+		return 1
+	return 0
+
+//----------------------------------------------Virology----------------------------------------------------
+
+//Vaccine
+/datum/centcomm_order/department/medical/vaccine
+	request_consoles_to_notify = list(
+		"Chief Medical Officer's Desk",
+		"Virology",
+		)
+	var/required_vac
+
+/datum/centcomm_order/department/medical/vaccine/New()
+	..()
+	requested = list(
+		/obj/item/weapon/reagent_containers/glass/beaker/vial = 1
+	)
+	name_override = list(
+		/obj/item/weapon/reagent_containers/glass/beaker/vial = "Vial of Vaccine"
+	)
+	var/difficulty = rand(1,4)
+	switch (difficulty)
+		if (1)
+			required_vac = pick(blood_antigens)
+			worth = 200
+		if (2)
+			required_vac = pick(common_antigens)
+			worth = 400
+		if (3)
+			required_vac = pick(rare_antigens)
+			worth = 800
+		if (4)
+			required_vac = pick(alien_antigens)
+			worth = 1600
+	extra_requirements = "Must contain [required_vac] antibodies."
+
+/datum/centcomm_order/department/medical/vaccine/ExtraChecks(var/obj/item/weapon/reagent_containers/glass/beaker/vial/V)
+	if (!istype(V))
+		return 0
+	for(var/datum/reagent/R in V.reagents?.reagent_list)
```

Comment:
```suggestion
	for(var/datum/reagent/vaccine/vaccine in V.reagents?.reagent_list)
```
and then you can get rid of the type check and cast.
you could also just use `get_reagent()` or `locate()`